pandas/core/indexes/base.py

```diff
@@ -4197,17 +4206,19 @@ def _get_leaf_sorter(labels):
                 mask = new_lev_codes != -1
                 if not mask.all():
                     new_codes = [lab[mask] for lab in new_codes]
-                    left_indexer = left_indexer[mask]
+                    # error: Value of type "Optional[ndarray]" is not indexable
+                    left_indexer = left_indexer[mask]  # type: ignore[index]
```

Comment:
np.arange resolves to `Any`.  https://github.com/numpy/numpy/blob/main/numpy/__init__.pyi

as a result `left_indexer` is not narrowed with `left_indexer = np.arange(len(left), dtype=np.intp)` (L4215 after merge master)

so I think in this case better to cast than add an ignore. (We have `warn_redundant_casts = True` in setup.cfg) so should be aware when the cast is no longer required)

Reply:
updated + green